applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp

```diff
@@ -553,6 +553,24 @@ array_1d<double, 3> FluidElement<TElementData>::GetAtCoordinate(
     return result;
 }
 
+template <class TElementData>
+Matrix FluidElement<TElementData>::GetAtCoordinate(
```

Comment:
The return type must be that one in the `NodalTensorData` (i.e. `BoundedMatrix<double,Dim,Dim>`).

Reply:
Are you sure you can define as `BoundedMatrix<double,Dim,Dim>`. I think you should define the return type as `BoundedMatrix<double,3,3>` or `BoundedMatrix<double,2,2>`.

Reply:
For this reason I think it is better to define the return type as `Matrix`

Reply:
I'm pretty sure you can as `Dim` is known at compile time. What might be required is to explicitly instantiate the method for 2 and 3. My point is that, as we discussed before, `Matrix` is much more inefficient...

Reply:
That's the point, I have tried to do a explicit instantiation for 2 and 3 but the compiler throws me an error which says that the method cannot be overloaded, maybe because only differ in the return type.

Reply:
Done.